app/services/role_setter.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class RoleSetter
+  def initialize(user_params:, provider:)
+    @user_params = user_params
+    @provider = provider
+    @roles_mappers = SettingGetter.new(setting_name: 'RoleMapping', provider:).call
+  end
+
+  def call
+    email = @user_params[:email]
+
+    matched_rule = if @roles_mappers
+                     rules = @roles_mappers.split(',').map { |rule| rule.split('=') }
+
+                     rules.find { |rule| email.ends_with? rule.second if rule.second }
+                   end
+
+    role = Role.find_by(name: matched_rule&.first) || Role.find_by(name: 'User')
```

Comment:
I would move this to a private method inside the service (after making the changes above)